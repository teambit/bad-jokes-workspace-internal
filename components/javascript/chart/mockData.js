const firstDataset = {
    labels: ['A', 'B', 'C'],
    datasets: [
      {
        data: [300, 50, 300],
        backgroundColor: ['#FF6384', '#36A2EB', '#FFCE56'],
        hoverBackgroundColor: ['#FF6384', '#36A2EB', '#FFCE56']
      }
    ]
  };
  
  const secondDataset = {
    labels: ['January', 'February', 'March', 'April', 'May', 'June', 'July'],
    datasets: [
      {
        label: 'First Dataset',
        data: [65, 59, 80, 81, 56, 55, 40],
        fill: false,
        borderColor: '#4bc0c0'
      },
      { 
        label: 'Second Dataset',
        data: [28, 48, 40, 19, 86, 27, 90],
        fill: false,
        borderColor: '#565656'
      }
    ]
  };
  
  const secondOptions = {
    title: {
      display: true,
      text: 'My Title',
      fontSize: 16
    },
    legend: {
      position: 'bottom'
    }
  };

  const mockData = {
    firstDataset, 
    secondDataset, 
    secondOptions
  };

  export default mockData;
  