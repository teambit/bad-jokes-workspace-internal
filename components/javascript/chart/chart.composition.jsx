import React from 'react';
import { Chart } from './chart';
import mockData from './mockData';

export const PieChartBasic = () => {
  return (
    <div style={{ width: 400 }}>
      <Chart type='pie' data={mockData.firstDataset} />
    </div>
  )
}

export const LineChartBasic = () => {
  return (
    <div style={{ width: 400 }}>
      <Chart type='line' data={mockData.secondDataset} options={mockData.secondOptions} />
    </div>
  )
}