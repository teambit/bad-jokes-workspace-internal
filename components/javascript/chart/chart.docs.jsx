import React from 'react';
import { Chart } from './chart';
import { Section } from '@teambit/documenter.ui.section';
import { ThemeContext } from '@teambit/documenter.theme.theme-context';
import { LinkedHeading } from '@teambit/documenter.ui.linked-heading';
import { List } from '@teambit/documenter.ui.list';
import { Separator } from '@teambit/documenter.ui.separator';

export default function Overview() {
  return (
    <ThemeContext> 
      <>
        <Section>
          <LinkedHeading link="guidelines">Guidelines</LinkedHeading>
          <List spacing="lg">
            {[
              `Chart components are based on Charts.js, an open source HTML5 based charting library.`,
              `This is an example of a js component being developed with the react-babel env`
            ]}
          </List>
        </Section>
        <Separator />
      </>
    </ThemeContext>
  );
};


Overview.abstract = 'A responsive chart by Primereasct, based on Chart.js';

Overview.labels = ['react', 'javascript', 'chart'];


const data1 = {
  labels: ['A', 'B', 'C'],
  datasets: [
    {
      data: [300, 50, 300],
      backgroundColor: ['#FF6384', '#36A2EB', '#FFCE56'],
      hoverBackgroundColor: ['#FF6384', '#36A2EB', '#FFCE56']
    }
  ]
};

const data2 = {
  labels: ['January', 'February', 'March', 'April', 'May', 'June', 'July'],
  datasets: [
    {
      label: 'First Dataset',
      data: [65, 59, 80, 81, 56, 55, 40],
      fill: false,
      borderColor: '#4bc0c0'
    },
    { 
      label: 'Second Dataset',
      data: [28, 48, 40, 19, 86, 27, 90],
      fill: false,
      borderColor: '#565656'
    }
  ]
};

const options2 = {
  title: {
    display: true,
    text: 'My Title',
    fontSize: 16
  },
  legend: {
    position: 'bottom'
  }
};


Overview.examples = [
  {
    scope: {
      Chart,
    },
    title: 'Using the Chart component',
    description: 'Basic pie chart.',
    jsx: 
    (
        <div style={{ width: 400 }}>
            <Chart type='pie' data={data1} />
        </div>)
  },
  {
    scope: {
      Chart,
    },
    description: 'Basic line chart.',
    jsx: 
    (
        <div style={{ width: 400 }}>
            <Chart type='line' data={data2} options={options2} />
        </div>)
  },
];


