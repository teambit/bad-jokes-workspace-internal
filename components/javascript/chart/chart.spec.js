import React from 'react';
import {Chart} from './chart';
import mockData from './mockData';

it('renders correctly', () => {
    const element = <Chart type='pie' data={mockData.firstDataset}/>;
    expect(element).toMatchSnapshot();
  });