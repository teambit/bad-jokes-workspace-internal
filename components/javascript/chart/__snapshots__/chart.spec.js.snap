// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Chart
  className={null}
  data={
    Object {
      "datasets": Array [
        Object {
          "backgroundColor": Array [
            "#FF6384",
            "#36A2EB",
            "#FFCE56",
          ],
          "data": Array [
            300,
            50,
            300,
          ],
          "hoverBackgroundColor": Array [
            "#FF6384",
            "#36A2EB",
            "#FFCE56",
          ],
        },
      ],
      "labels": Array [
        "A",
        "B",
        "C",
      ],
    }
  }
  height={null}
  id={null}
  options={null}
  style={null}
  type="pie"
  width={null}
/>
`;
